### PR TITLE
Add Land Transaction Tax, Land and Buildings Transaction Tax and business rates

### PR DESCRIPTION
**Pull request by nikhilwoodruff**
Adds logic and parameters for Land Transaction Tax and LBTT (Wales and Scotland's Stamp Duty equivalents, respectively).

### CODE REVIEW

File: openfisca_uk/parameters/revenue_scotland/lbtt/non_residential.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+description: Rates on residential property puchases
```
Comment (MaxGhenis) [inline openfisca_uk/parameters/revenue_scotland/lbtt/non_residential.yaml:1]:
```suggestion
description: Rates on Scotland's non-residential property purchases
```

File: openfisca_uk/parameters/revenue_scotland/lbtt/residential/rate.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+description: Rates on residential property puchases
```
Comment (MaxGhenis) [inline openfisca_uk/parameters/revenue_scotland/lbtt/residential/rate.yaml:1]:
```suggestion
description: Rates on Scotland's residential property purchases
```

File: openfisca_uk/tests/policy/baseline/wra/land_transaction_tax/land_transaction_tax.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  output:
+    land_transaction_tax: 331_200
+- name: Non-residential property - very high value after LTT holiday
+  period: 2022
+  input:
+    non_residential_property_purchased: 2_500_000
+    country: WALES
+  output:
+    land_transaction_tax: 127_750
+
```
Comment (MaxGhenis) [inline openfisca_uk/tests/policy/baseline/wra/land_transaction_tax/land_transaction_tax.yaml:44]: extra line

File: openfisca_uk/variables/finance/income.py [hunk 4]
```diff
@@ -205,8 +205,10 @@ class hbai_household_net_income(Variable):
     def formula(household, period, parameters):
         gross_income = household("household_gross_income", period)
         tax = household("household_tax", period)
-        ignored_taxes = household("expected_stamp_duty", period) + household(
-            "corporate_tax_incidence", period
+        ignored_taxes = (
```
Comment (MaxGhenis) [inline openfisca_uk/variables/finance/income.py:208]: can this use `add`?

File: openfisca_uk/variables/finance/tax/tax.py [hunk 5]
```diff
@@ -27,6 +27,7 @@ def formula(household, period):
         personal_taxes = household.sum(household.members("tax", period))
         household_taxes = (
             household("expected_stamp_duty", period)
+            + household("expected_land_transaction_tax", period)
```
Comment (MaxGhenis) [inline openfisca_uk/variables/finance/tax/tax.py:30]: same - `add`

File: openfisca_uk/variables/revenue_scotland/lbtt.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
+from openfisca_uk.model_api import *
+
+
+class lbtt_on_residential_property(Variable):
```
Comment (MaxGhenis) [inline openfisca_uk/variables/revenue_scotland/lbtt.py:4]: so i prefer lowercase acronyms but we should probably be consistent for now and switch them all over if we decide to do that

File: openfisca_uk/variables/revenue_scotland/lbtt.py [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    label = "Land and Buildings Transaction Tax on residential property"
+    documentation = (
+        "Tax charge from purchase or rental of residential property"
+    )
+    entity = Household
+    definition_period = YEAR
+    value_type = float
+    unit = "currency-GBP"
+
+    def formula(household, period, parameters):
```
Comment (MaxGhenis) [inline openfisca_uk/variables/revenue_scotland/lbtt.py:14]: is there a way to reduce code duplication between this and sdlt?

File: openfisca_uk/variables/revenue_scotland/lbtt.py [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+        lbtt = parameters(period).revenue_scotland.lbtt
+        # Tax on non-residential purchases
+        price = household("non_residential_property_purchased", period)
+        non_residential_purchase_tax = lbtt.non_residential.calc(price)
+        # Tax on non-residential rents
+        cumulative_rent = household("cumulative_non_residential_rent", period)
+        rent = household("non_residential_rent", period)
```
Comment (MaxGhenis) [inline openfisca_uk/variables/revenue_scotland/lbtt.py:65]: what would a household's non-residential rent include? is this in the microdata?

File: openfisca_uk/variables/revenue_scotland/lbtt.py [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+
+    def formula(household, period):
+        country = household("country", period)
+        countries = country.possible_values
+        return country == countries.SCOTLAND
+
+
+class land_and_buildings_transaction_tax(Variable):
+    label = "Land and Buildings Transaction Tax"
+    documentation = "Total tax liability for LBTT"
```
Comment (MaxGhenis) [inline openfisca_uk/variables/revenue_scotland/lbtt.py:90]:
```suggestion
    documentation = "Total tax liability for Scotland's LBTT"
```

File: openfisca_uk/variables/revenue_scotland/lbtt.py [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+    entity = Household
+    definition_period = YEAR
+    value_type = float
+    unit = "currency-GBP"
+
+    def formula(household, period):
+        return household("lbtt_on_residential_property", period) + household(
```
Comment (MaxGhenis) [inline openfisca_uk/variables/revenue_scotland/lbtt.py:97]: `add`

File: openfisca_uk/variables/wealth/corporate.py [hunk 11]
```diff
@@ -21,14 +39,7 @@ class corporate_tax_incidence(Variable):
...
-        total_change = household.state(
-            "corporate_stamp_duty_revenue_change", period
-        )
-        total_wealth = (
-            household("corporate_wealth", period)
-            * household("household_weight", period)
-        ).sum()
-        percentage_change = where(
-            total_wealth > 0, total_change / total_wealth, 0
+        total_change = household("corporate_stamp_duty", period) - household(
```
Comment (MaxGhenis) [inline openfisca_uk/variables/wealth/corporate.py:42]:
```suggestion
        return household("corporate_stamp_duty", period) - household(
```

File: openfisca_uk/variables/wealth/land.py [hunk 12]
```diff
@@ -11,22 +11,36 @@ class owned_land(Variable):
...
+    documentation = (
+        "Estimated total land value (directly and indirectly owned)"
+    )
+    entity = Household
+    definition_period = YEAR
+    value_type = float
+    unit = "currency-GBP"
+
+    def formula(household, period):
+        return household("household_land_value", period) + household(
```
Comment (MaxGhenis) [inline openfisca_uk/variables/wealth/land.py:24]: `add`

File: openfisca_uk/variables/wra/land_transaction_tax.py [hunk 13]
```diff
@@ -0,0 +1,94 @@
+from openfisca_uk.model_api import *
+
+
+class ltt_liable(Variable):
+    label = "Liable for Land Transaction Tax"
+    documentation = (
+        "Whether the household is liable to pay the Land Transaction Tax"
```
Comment (MaxGhenis) [inline openfisca_uk/variables/wra/land_transaction_tax.py:7]:
```suggestion
        "Whether the household is liable to pay the Wales Land Transaction Tax"
```

File: openfisca_uk/variables/wra/land_transaction_tax.py [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+    definition_period = YEAR
+    value_type = bool
+
+    def formula(household, period):
+        country = household("country", period)
+        return country == country.possible_values.WALES
+
+
+class ltt_on_residential_property(Variable):
+    label = "LTT on residential property"
```
Comment (MaxGhenis) [inline openfisca_uk/variables/wra/land_transaction_tax.py:19]: This PR is inconsistent on acronyms in labels vs documentation. I'd prefer keeping acronyms out of labels and putting them inside parens in the documentation. I'd also suggest putting the country in the documentation.

**Comment by nikhilwoodruff** [comment]
@MaxGhenis I've just added the business rates variables/parameters and added all property taxes to the documentation. Here's the new page: 
[Property Taxes — OpenFisca-UK documentation.pdf](https://github.com/PolicyEngine/openfisca-uk/files/7745113/Property.Taxes.OpenFisca-UK.documentation.pdf)

**Pull request merged**